Domain.Sql/DatabaseExtensions.cs

```diff
@@ -101,9 +97,20 @@ public static bool IsAzureDatabase(this IDbConnection connection)
             {
                 InitialCatalog = "master"
             };
+            if (sqlAzureDatabaseProperties == null)
```

Comment:
If you're allowing the `sqlAzureDatabaseProperties` parameter to be null, you can signal that in the signature by making it optional.